packages/js/product-editor/src/components/header/header.tsx

```diff
@@ -91,6 +97,8 @@ export function Header( { productId, productName }: HeaderProps ) {
 						? __( 'Add', 'woocommerce' )
 						: __( 'Save', 'woocommerce' ) }
 				</Button>
+				<WooProductHeaderItem.Slot />
```

Comment:
I think another option is something in the middle, where we use an approach like `ActionItem`, but for the header.

For example:

`<WooHeaderItem.Slot name="product" fillProps={ ... } />`

Reply:
Right, I wanted to avoid that originally due to the potential problem you mention, with fills ending up in unintended slots if they're rendered at the wrong time. 

That said, altering the `name` is a decent enough solution to avoid this. I've made this change in 212c427e367510da881bb77058e647fb14fe2f40.